### ISSUE TITLE
Object set to null in Session becomes "null" in Session when deserialized

### ISSUE THREAD

**Issue by judowalker** [MarkCBB/MongoDB-ASP.NET-Session-State-Store#7]
When I set an object in Session to null, the RawSerialization is serializing the value as `"null"` but when it's deserialized, it's not setting the value back to null.
https://github.com/MarkCBB/MongoDB-ASP.NET-Session-State-Store/blob/master/MongoSessionStateStore/Serialization/RawSerialization.cs#L36

**Comment by judowalker** [comment]
@MarkCBB we have a deployment next week and it would be nice if this was fixed asap so that we don't have to postpone our deployment.

**Comment by MarkCBB** [comment]
Hello,

There is a test case working as expected. [This test case](https://github.com/MarkCBB/MongoDB-ASP.NET-Session-State-Store/blob/master/TestApplication_RAWSerialization/Controllers/HomeController.cs#L159) set and get a null value and [this other test](https://github.com/MarkCBB/MongoDB-ASP.NET-Session-State-Store/blob/master/TestApplication_RAWSerialization/Controllers/HomeController.cs#L168) case get a null value in an independent request.

And, debugging both you'll see the null value:
![image](https://cloud.githubusercontent.com/assets/8301580/12917692/2bf0b2da-cf3b-11e5-901d-92984c95eb96.png)

![image](https://cloud.githubusercontent.com/assets/8301580/12917698/3089a2d4-cf3b-11e5-91f8-fc2eafee5773.png)

But indeed the value that you can see in the database is the "null" string serialized ("AAEAAAD/////AQAAAAAAAAAGAQAAAARudWxsCw==")

An improvement could be instead of to serialize the string "null" using the same serialization for everything, insert a null value in the document. **This means that I've to modify the Serialize and Deserialize methods of RawSerialization.**

Do you really want this?

**Comment by MarkCBB** [comment]
This commit contains the detailed modification in a new branch. Serialized values are serialized as null values in serialization based in RAW. All test cases have been passed.

If you are agree I'll merge the branch issue_7.

About to finish before, ok, I'll try!

Here how looks like the document:

```
{
    "_id" : "xjdwkfn3mh33de3rewjo3egd-RAWTestApplication",
    "SessionID" : "xjdwkfn3mh33de3rewjo3egd",
    "ApplicationName" : "RAWTestApplication",
    "Created" : ISODate("2016-02-09T14:58:11.094Z"),
    "Expires" : ISODate("2016-02-09T15:18:11.104Z"),
    "LockDate" : ISODate("2016-02-09T14:58:11.101Z"),
    "LockId" : 1,
    "Timeout" : 20,
    "Locked" : false,
    "SessionItemBSON" : [
        {
            "PersonNull" : null
        }
    ],
    "Flags" : 0
}
```

**Comment by judowalker** [comment]
That test doesn't actually test if the value is null because using `as Person` will return null if the cast fails. It's returning null because it cannot cast `"null"` into `Person`.

If you change the test to `Person personGet = (Person)Session["PersonNull"]`` you'll see that it fails.

`Session["PersonNull"] == null` would also return false.

**Comment by MarkCBB** [comment]
Indeed, well seen. I have not paid enough attention to `as` operator. Anyway the changes are developed in the issue_7 branch and I'm gonna merge it.
Also I'll modify the test case to make it more accurate.

**Comment by MarkCBB** [comment]
I've merged the changes in master branch. All test cases have been successfully passed.
**[I've uploaded a release candidate in nuget website](https://www.nuget.org/packages/MongoSessionStateStore/4.0.0-rc)** to enable you to start working in this version (takes sometime to be indexed and available for everyone).
If you find any issue please let me know!
Tomorrow I'll pack the definitive package, I'll test the packed solution with all versions of mongoDB driver, I'll update the documentation and finally I'll upload the final pack.
I'm working as fast as I can, hope that helps for your deadlines.

This issue is planned to be solved in release v4.0.0.

**Comment by judowalker** [comment]
Great, thanks for the quick response and fix.

**Comment by MarkCBB** [comment]
Published the v4.0.0!
I'll keep open this issue some days, let me know if something goes wrong.

**Comment by judowalker** [comment]
Thanks a lot. I will download the new package shortly.
